dot/network/README.md

```diff
@@ -0,0 +1,178 @@
+
+### Stream Multiplexing
+
+[Multiplexing](https://en.wikipedia.org/wiki/Multiplexing) allows multiple independent logical streams to share a common
+underlying transport medium, which amortizes the overhead of establishing new connections with peers in a P2P network.
+In particular, `libp2p` relies on "[stream multiplexing](https://docs.libp2p.io/concepts/stream-multiplexing/)", which
+uses logically distinct "paths" to route requests to the proper handlers. A familiar example of stream multiplexing
+exists in the TCP/IP stack, where unique port numbers are used to distinguish logically independent streams that share a
+common physical transport medium. Gossamer uses [Yamux](#yamux) for stream multiplexing.
+
+## Gossamer Network Protocols
+
+### `libp2p`-Integrated Protocols
+
+#### `ping`
+
+This is a simple liveness check [protocol](https://docs.libp2p.io/concepts/protocols/#ping) that peers can use to
+quickly see if another peer is online - it is
+[included](https://github.com/libp2p/go-libp2p/tree/master/p2p/protocol/ping) with the official Go implementation of
+`libp2p`.
+
+#### `identify`
+
+The [`identify` protocol](https://docs.libp2p.io/concepts/protocols/#identify) allows peers to exchange information
+about each other, most notably their public keys and known network addresses; like [`ping`](#ping), it is
+[included with `go-libp2p`](https://github.com/libp2p/go-libp2p/tree/master/p2p/protocol/identify).
+
+#### Noise
+
+As described above, [Noise](http://noiseprotocol.org/) provides `libp2p` with its
+[key distribution](#identity--key-management) capabilities. The Noise protocol is
+[well documented](http://cryptowiki.net/index.php?title=Noise_Protocol_Framework) and the Go implementation is
+maintained [under the official](https://github.com/libp2p/go-libp2p-noise) `libp2p` GitHub organization. Noise defines a
+number of [variables](http://cryptowiki.net/index.php?title=Noise_Protocol_Framework#Noise_Variables) and
+[handshake patterns](http://cryptowiki.net/index.php?title=Noise_Protocol_Framework#Handshake_patterns) that
+participants in a peer-to-peer network can use to establish message-passing channels with one another.
+
+#### Yamux
+
+[Yamux (Yet another Multiplexer)](https://github.com/hashicorp/yamux) is a Golang library for
+[stream-oriented multiplexing](#stream-multiplexing) that is maintained by [HashiCorp](https://www.hashicorp.com/) - it
+implements a well defined [specification](https://github.com/hashicorp/yamux/blob/master/spec.md). Gossamer uses
+[the official `libp2p` adapter](https://github.com/libp2p/go-libp2p-yamux) for Yamux.
+
+#### Kademlia
+
+[Kademlia](https://en.wikipedia.org/wiki/Kademlia) is a battle-tested
+[distributed hash table (DHT)](https://en.wikipedia.org/wiki/Distributed_hash_table) that defines methods for managing a
+dynamic list of peers that is constantly updated in order to make a P2P network more resilient and resistant to attacks.
```

Comment:
I'd clarify this by explicitly saying the DHT is used for peer discovery, and that each node will advertise themselves in the DHT, and to discover peers nodes will "walk" the DHT.